tests/testthat/test-step-call.R

```diff
@@ -161,3 +161,36 @@ test_that("unique is an alias for distinct", {
   dt <- lazy_dt(data.table(x = c(1, 1)))
   expect_equal(unique(dt), distinct(dt))
 })
+
+# drop_na ------------------------------------------------------------------
+
+test_that("empty call drops every row", {
+  tb <- tibble(x = c(1, 2, NA), y = c("a", NA, "b"))
+  step <- drop_na(lazy_dt(tb, "DT"))
+  expect_equal(show_query(step), expr(na.omit(DT)))
+  expect_equal(as_tibble(step), tb[1, ])
+})
+
+test_that("converts data.table to dtplyr_step", {
+  df <- data.table(x = c(1, 2, NA), y = c("a", NA, "b"))
+  expect_equal(class(drop_na(df)), c("dtplyr_step_call", "dtplyr_step"))
+})
+
+test_that("specifying (a) variables considers only that variable(s)", {
+  df <- lazy_dt(tibble(x = c(1, 2, NA), y = c("a", NA, "b")), "DT")
+  exp <- tibble(x = c(1, 2), y = c("a", NA))
+  step <- drop_na(df, x)
+  res <- as_tibble(step)
+  expect_equal(show_query(step), expr(na.omit(DT, cols = "x")))
+  expect_equal(res, exp)
+  exp <- tibble(x = c(1), y = c("a"))
+  res <- as_tibble(drop_na(df, x:y))
+  expect_equal(res, exp)
+})
+
+test_that("errors are raised", {
+  tb <- tibble(x = c(1, 2, NA), y = c("a", NA, "b"))
+  dt <- lazy_dt(tb, "DT")
+  expect_snapshot(collect(drop_na(dt, !! list())), error = TRUE)
```

Comment:
I'm not sure what this is supposed to test, and it looks like the `!!` is being evaluated too early. So I suggest just dropping it.